src/Internal/DependencyManagement/AbstractServiceProvider.php

```diff
@@ -61,6 +61,22 @@ protected function add_with_auto_arguments( string $class_name, $concrete = null
 		return $definition;
 	}
 
+	/**
+	 * Gets the class of a parameter.
+	 *
+	 * This method is a replacement for ReflectionParameter::getClass,
+	 * which is deprecated as of PHP 8.
+	 *
+	 * @param \ReflectionParameter $parameter The parameter to get the class for.
+	 *
+	 * @return \ReflectionClass|null The class of the parameter, or null if it hasn't any.
+	 */
+	private function get_class( \ReflectionParameter $parameter ) {
+		return $parameter->getType() && ! $parameter->getType()->isBuiltin()
+			? new \ReflectionClass( $parameter->getType()->getName() )
```

Comment:
I think we will need to change this as ReflectionType::getName() is not available in PHP 7.0 (https://travis-ci.org/github/woocommerce/woocommerce/jobs/736310447#L568)

Reply:
Ooops, good catch. Fixed!